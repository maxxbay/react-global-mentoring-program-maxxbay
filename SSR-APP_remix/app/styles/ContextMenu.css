.context-menu {
    position: absolute;
    background-color: #424242;
    border-radius: 4px;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.3);
    overflow: hidden;
}

.context-menu__item {
    display: block;
    padding: 12px 24px;
    cursor: pointer;
    text-align: left;
    width: 100%;
    border: none;
    background-color: #424242;
    color: #ffffff;
    font-size: 16px;
    line-height: 1.5;
}

.context-menu__item:hover {
    background-color: #616161;
}