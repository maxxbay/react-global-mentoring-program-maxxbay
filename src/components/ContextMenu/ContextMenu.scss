.context-menu {
    position: absolute;
    background-color: #424242;
    border-radius: 4px;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.3);
    overflow: hidden;

    &__item {
        display: block;
        padding: 12px 24px;
        cursor: pointer;
        text-align: left;
        width: 100%;
        border: none;
        background-color: #424242;
        color: #ffffff;
        font-size: 16px;
        line-height: 1.5;

        &:hover {
            background-color: #616161;
        }
    }
}