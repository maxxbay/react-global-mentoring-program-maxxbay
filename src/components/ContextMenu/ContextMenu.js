import React from 'react';
import './ContextMenu.scss';
import PropTypes from 'prop-types';

const ContextMenu = ({ show, onClose, onEdit, onDelete }) => {
  const handleEdit = e => {
    onEdit();
    onClose(e);
  };

  const handleDelete = e => {
    onDelete();
    onClose(e);
  };

  return (
    show && (
      <div className="context-menu">
        <button className="context-menu__item" onClick={handleEdit}>
          Edit
        </button>
        <button className="context-menu__item" onClick={handleDelete}>
          Delete
        </button>
      </div>
    )
  );
};
ContextMenu.propTypes = {
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};
export default ContextMenu;
