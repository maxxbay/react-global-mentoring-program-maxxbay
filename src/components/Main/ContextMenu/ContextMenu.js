import React from 'react';
import './ContextMenu.scss';

const ContextMenu = ({ show, onClose, onEdit, onDelete }) => {
  const handleEdit = e => {
    e.stopPropagation();
    onEdit();
    onClose();
  };

  const handleDelete = e => {
    e.stopPropagation();
    onDelete();
    onClose();
  };

  return (
    show && (
      <div className="context-menu">
        <button className="context-menu__item" onClick={handleEdit}>
          Edit
        </button>
        <button className="context-menu__item" onClick={handleDelete}>
          Delete
        </button>
      </div>
    )
  );
};

export default ContextMenu;
