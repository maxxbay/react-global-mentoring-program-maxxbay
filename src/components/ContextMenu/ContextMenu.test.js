import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ContextMenu from './ContextMenu';

describe('ContextMenu component', () => {
  const onCloseMock = jest.fn();
  const onEditMock = jest.fn();
  const onDeleteMock = jest.fn();

  beforeEach(() => {
    onCloseMock.mockClear();
    onEditMock.mockClear();
    onDeleteMock.mockClear();
  });

  it('renders correctly when show is true', () => {
    const { asFragment } = render(
      <ContextMenu
        show
        onClose={onCloseMock}
        onEdit={onEditMock}
        onDelete={onDeleteMock}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render when show is false', () => {
    const { queryByTestId } = render(
      <ContextMenu
        show={false}
        onClose={onCloseMock}
        onEdit={onEditMock}
        onDelete={onDeleteMock}
      />
    );
    expect(queryByTestId('context-menu')).toBeNull();
  });

  it('calls onEdit and onClose when edit button is clicked', () => {
    const { getByText } = render(
      <ContextMenu
        show
        onClose={onCloseMock}
        onEdit={onEditMock}
        onDelete={onDeleteMock}
      />
    );
    fireEvent.click(getByText('Edit'));

    expect(onEditMock).toHaveBeenCalledTimes(1);
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });

  it('calls onDelete and onClose when delete button is clicked', () => {
    const { getByText } = render(
      <ContextMenu
        show
        onClose={onCloseMock}
        onEdit={onEditMock}
        onDelete={onDeleteMock}
      />
    );
    fireEvent.click(getByText('Delete'));

    expect(onDeleteMock).toHaveBeenCalledTimes(1);
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });
});
