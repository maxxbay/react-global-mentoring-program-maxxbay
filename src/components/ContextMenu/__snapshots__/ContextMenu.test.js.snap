// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu component renders correctly when show is true 1`] = `
<div>
  <div
    class="context-menu"
  >
    <button
      class="context-menu__item"
    >
      Edit
    </button>
    <button
      class="context-menu__item"
    >
      Delete
    </button>
  </div>
</div>
`;
